desc/compute/utils.py

```diff
@@ -820,22 +1194,29 @@ def surface_min(grid, x, surface_label="rho"):
         Collocation grid containing the nodes to evaluate at.
     x : ndarray
         Quantity to find min.
+        The array should have size grid.num_nodes.
     surface_label : str
         The surface label of rho, theta, or zeta to compute min over.
 
     Returns
     -------
     mins : ndarray
         Minimum of x over each surface in grid.
+        The returned array has the same shape as the input.
 
     """
-    _, unique_idx, inverse_idx, _, _, _ = _get_grid_surface(grid, surface_label)
+    unique_size, inverse_idx, _, _ = _get_grid_surface(grid, surface_label)
     inverse_idx = jnp.asarray(inverse_idx)
     x = jnp.asarray(x)
-    mins = jnp.inf * jnp.ones(unique_idx.size)
+    mins = jnp.full(unique_size, jnp.inf)
 
     def body(i, mins):
         mins = put(mins, inverse_idx[i], jnp.minimum(x[i], mins[inverse_idx[i]]))
         return mins
 
-    return fori_loop(0, len(inverse_idx), body, mins)
+    mins = fori_loop(0, inverse_idx.size, body, mins)
+    # The above implementation was benchmarked to be more efficient, after jit
+    # compilation, than the alternative given in the two lines below.
+    # masks = inverse_idx == jnp.arange(unique_size)[:, jnp.newaxis]  # noqa: E501,E800
+    # mins = jnp.amin(x[jnp.newaxis, :], axis=1, initial=jnp.inf, where=masks)  # noqa: E501,E800
```

Comment:
Do we need to keep this record in the code?

Reply:
Could probably eventually move to some sort of design doc or dev guide but I think it's fine to leave it in.

Reply:
We don't need to. Some of my other comments may be redundant too; I mainly included them to make reviewing easier. I'll remove this comment from my add_all_limits pr. If there are other cosmetic changes just mark them and I'll do the same.